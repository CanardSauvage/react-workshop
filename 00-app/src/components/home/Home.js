/**
 * Home Component
 *
 */

import React, { Component } from 'react';
import { Link } from 'react-router';

//
// Default values and utility functions
//

import { getUniverseURL, postUniverseURL, postUniverseConfig } from "../../settings/settings";
import { fetch, post } from "../../utility/utility";

//
// Styles
//

import "../../assets/sass/components/_dashboard.scss";

//
// Define Components
//

const Items = ({ items }) => {
  const universes = items.map((item, index) => {
    return (
      <Link
        to={`/universe/${item.id}`}
        className="dash__items__item-universe"
        key={index}
      >
        <div className="dash__items__item-universe__image"></div>
        <div className="dash__items__item-universe__title">
          {item.name}
        </div>
      </Link>
    )
  });

  return (
    <div className="dash__items">
      {universes}
    </div>
  );
};

class Home extends Component {

  //
  // Set Default State
  //

  constructor(props) {
    super(props);
    this.state = {
      universes: [],
      showInput: false
    };
  }

  //
  // Methods
  //

  handleAddUniverse() {
    this.setState({
      showInput: !this.state.showInput
    });
  }

  handleSubmit() {
    const universeName = this.refs.universeName.value;
    const universeMaxSize = this.refs.universeMaxSize.value;

    // Simple naive form validation, submit only if validated
    if (universeName.length > 0 && !isNaN(parseInt(universeMaxSize, 10))) {

      // Construct new universe

      const newUniverse = {
        name: universeName,
        maxSize: universeMaxSize
      }

      // Put new universe on state object

      this.setState({
        universes: [...this.state.universes, newUniverse],
        showInput: false
      });

      // POST data to server

      post(postUniverseURL, newUniverse, postUniverseConfig)
        .then((response) => {
          console.log("Post Success");
        })
        .catch((error) => {
          console.log(error);
        });
    }
  }

  //
  // Lifecycle Methods
  //

  componentDidMount() {
    // GET universes
    fetch(getUniverseURL)
      .then((response) => {
        this.setState({
          universes: response.data.universes
        })
      })
      .catch((error) => {
        console.log(error);
      });
  }

  //
  // Render
  //

  render() {
    return (
      <div className="dash">
        <div className="dash__titlebar">
          <div className="dash__titlebar-item -stretch">
            Universes
            <span className="dash__titlebar-circle">
              {this.state.universes.length > 0 ? this.state.universes.length : ""}
            </span>
          </div>
          <div className="dash__titlebar-item">
            <a
              onClick={this.handleAddUniverse.bind(this)}
              className={this.state.showInput ? "dash__titlebar-link dash__titlebar-link--active" : "dash__titlebar-link"}
            >
              Add Universe
            </a>
          </div>
        </div>
        <Items items={this.state.universes} />
        {/*
          Render depending on state by using inline JS
        */}
        {
          (() => {
            if (this.state.showInput) {
              return (
                <div className="dash__overlay">
                  <h1 className="dash__overlay-title">Neues Universum hinzuf√ºgen</h1>
                  <input ref="universeName" type="text" className="input-text" placeholder="Name" />
                  <input ref="universeMaxSize" type="text" className="input-text" placeholder="Anzahl" />
                  <a onClick={this.handleSubmit.bind(this)} className="btn">Abschicken </a>
                </div>
              )
            }
          })()
        }
      </div>
    );
  }
}

export default Home;
