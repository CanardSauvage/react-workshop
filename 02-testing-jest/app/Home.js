import React from 'react';

export default class Home extends React.Component {
    render() {
        return (
            <div className="home">
                <h1>23 + 19</h1>
                <h2>= {this.props.answer}</h2>
            </div>
        );
    }
}
