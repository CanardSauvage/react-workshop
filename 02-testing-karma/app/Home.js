import React from 'react';

export default class Home extends React.Component {
    render() {
        return (
            <div className="home">
                <h1>23 + 19</h1>
                <h2 className="home__answer">= {this.props.answer}</h2>
            </div>
        );
    }
}
