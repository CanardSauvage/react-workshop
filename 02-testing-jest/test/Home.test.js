import React from 'react';
import renderer from 'react-test-renderer';
import Home from '../app/Home.js';

it('renders correctly', () => {
    const tree = renderer.create(
        <Home answer="42" />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
