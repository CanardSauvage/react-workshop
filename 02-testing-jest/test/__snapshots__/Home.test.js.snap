exports[`test renders correctly 1`] = `
<div
  className="home">
  <h1>
    23 + 19
  </h1>
  <h2>
    = 
    42
  </h2>
</div>
`;
